### DISCUSSION TITLE
Programmatically uploading (geo)JSON files throws mimetype error

### DISCUSSION THREAD

**Discussion by stefgootzen** [payloadcms/payload#2204] [Roadmap]
# Bug Report
Calling this: 

```javascript
const filePath = '[...]/filename.geojson';

await payload.create({
    collection: 'geofiles',
    filePath: tmpFilePath,
    data: {}
  });
```

results in the following error:
```
TypeError: Cannot read properties of undefined (reading 'mime')
    at getFileByPath ([...]/node_modules/payload/src/uploads/getFileByPath.ts:16:34)
    at async createLocal ([...]/node_modules/payload/src/collections/operations/local/create.ts:65:21)
```

I'm using Payload 1.6.6. 

## Possible cause
I believe the problem lies in the the following code `const mimetype = fromFile(filePath);` [(link)](https://github.com/payloadcms/payload/blob/b0b82e1f57c1c59a06be3a4747ec1f501c2fe27d/src/uploads/getFileByPath.ts#L9) which ultimately in the [file-type](https://github.com/sindresorhus/file-type) package does this `new FileTypeParser().parse(tokenizer);`. 

`FileTypeParser.parse()` [(link)](https://github.com/sindresorhus/file-type/blob/9b24116d7b386141978d51c4cc582cee010cc839/core.js#L76) does not support (geo)JSON.

The `file-type` states _This package is for detecting binary-based file formats, not text-based formats like .txt, .csv, .svg, etc._, does this mean that the payload upload system also only supports binary-based file formats?

**Comment by jmikrut** [comment]
Hey @stefgootzen this is very interesting. I have never run into this myself but I can see how this package would definitely cause some issues with text-based files. 

We can likely swap out this dependency to support text-based files. I will convert this to a Discussion and then add it to our Roadmap. 

If you'd like to help here, we would happily accept a PR. Looks like the package is only used in 3 places within our code so if we can find a suitable replacement, we can certainly swap it out.

Otherwise, we should be able to get to this in our Roadmap Priority 2 (next quarter).

**Comment by mcimam** [comment]
Confirm it, I also encounter this error when uploading csv file via local api.
Any possible solution for now?

**Comment by stefgootzen** [comment]
Try reading the file first, something like this:

```javascript
import fs from "fs";

const tmpFilePath = 'filePathHere'
const fileBuffer = fs.readFileSync(tmpFilePath);

const geojson = await payload.create({
  collection: 'geofiles',
  file: {
    data: fileBuffer,
    mimetype: 'application/geo+json',
    name: filename,
    size: fileBuffer.byteLength
  },
  data: {}
});
return geojson.id;
```

**Comment by JarrodMFlesch** [comment]
I just tested this on latest and it worked as expected

```ts
const geoJsonPath = path.resolve(dirname, './sample.geojson')
const fileBuffer = fs.readFileSync(geoJsonPath)

const doc = await payload.create({
  collection: noRestrictFileTypesSlug as CollectionSlug,
  data: {},
  file: {
    data: fileBuffer,
    mimetype: 'application/geo+json',
    name: 'sample.geojson',
    size: fileBuffer.length,
  },
})
```
